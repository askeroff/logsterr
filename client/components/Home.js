import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Layout from './layout/Layout';
import Dashboard from './dashboard/Index';
import Spinner from './layout/Spinner';

class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loggedIn: false,
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.user.loggedIn === true) {
      this.setState({ loggedIn: true });
    } else if (nextProps.user.loggedIn === false) {
      this.props.history.push('/login');
    }
  }

  shouldRender() {
    if (this.state.loggedIn) {
      return <Dashboard />;
    }
    return <Spinner />;
  }

  render() {
    return (
      <Layout>
        {this.shouldRender()}
      </Layout>
    );
  }
}

Home.propTypes = {
  user: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  user: state.user,
});

export default connect(mapStateToProps)(Home);
