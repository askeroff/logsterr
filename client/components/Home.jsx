"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
exports.__esModule = true;
var React = require("react");
var react_redux_1 = require("react-redux");
var Layout_1 = require("./layout/Layout");
var Index_1 = require("./dashboard/Index");
var Spinner_1 = require("./layout/Spinner");
var LandingPage_1 = require("./LandingPage");
var Home = /** @class */ (function (_super) {
    __extends(Home, _super);
    function Home() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.shouldRender = function () {
            var hasHistory = Object.prototype.hasOwnProperty.call(_this.props.history, 'push');
            if (_this.props.user.loggedIn) {
                return (<Layout_1.default showToGuests>
          <Index_1.default />
        </Layout_1.default>);
            }
            else if (_this.props.user.loggedIn === false && hasHistory) {
                return (<Layout_1.default showToGuests contentClass="landing-wrapper">
          <LandingPage_1.default />
        </Layout_1.default>);
            }
            return (<Layout_1.default showToGuests>
        <Spinner_1.default />
      </Layout_1.default>);
        };
        return _this;
    }
    Home.prototype.render = function () {
        return this.shouldRender();
    };
    Home.defaultProps = { history: {}, user: {} };
    return Home;
}(React.Component));
var mapStateToProps = function (state) { return ({
    user: state.user
}); };
exports["default"] = react_redux_1.connect(mapStateToProps)(Home);
