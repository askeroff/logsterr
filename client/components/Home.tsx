import * as React from 'react';
import { connect } from 'react-redux';
import Layout from './layout/Layout';
import Dashboard from './dashboard/Index';
import Spinner from './layout/Spinner';
import { IUser } from '../types';
import LandingPage from './LandingPage';

interface Props {
  user: IUser,
  history: any
}

class Home extends React.Component<Props, any> {
  static defaultProps = { history: {}, user: {} };

  shouldRender = () => {
    const hasHistory = Object.prototype.hasOwnProperty.call(
      this.props.history,
      'push'
    );
    if (this.props.user.loggedIn) {
      return (
        <Layout showToGuests>
          <Dashboard />
        </Layout>
      );
    } else if (this.props.user.loggedIn === false && hasHistory) {
      return (
        <Layout showToGuests contentClass="landing-wrapper">
          <LandingPage />
        </Layout>
      );
    }
    return (
      <Layout showToGuests>
        <Spinner />
      </Layout>
    );
  };

  render() {
    return this.shouldRender();
  }
}

const mapStateToProps = state => ({
  user: state.user
});

export default connect(mapStateToProps)(Home);
