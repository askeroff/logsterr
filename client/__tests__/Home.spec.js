import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Home from '../components/Home';

Enzyme.configure({ adapter: new Adapter() });

test('Home Component renders correctly', () => {
  const component = shallow(<Home />);
  expect(component).toMatchSnapshot();
});
