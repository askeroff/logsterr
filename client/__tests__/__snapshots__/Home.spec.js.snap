// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders correctly 1`] = `
<Connect(Layout)>
  <h1>
    Home Page
  </h1>
</Connect(Layout)>
`;
